qcodes/utils/dataset/doNd.py

```diff
@@ -143,18 +154,27 @@ def do1d(
         write_period: The time after which the data is actually written to the
             database.
         additional_setpoints: A list of setpoint parameters to be registered in
-            the measurement but not scanned.
+            the measurement but not scanned. TODO it should be enforced that
+            these are scalar parameters
```

Comment:
How should we handle these in the shape of the data. As a new dimension with shape (1,) or just ignore?

Reply:
that's a good question:
* if we ignore them, will loading that data work nicely for the main setpoint?
* if we don't ignore, will the resulting data have these extra dimensions for each additional setpoint? is this how users like their data when they use the additional_setpoints?

i think it's best to ask some fans of additional_setpoints argument for input.

we can start by ignoring because wew can extend it to (1,) if needed without backwards compatibility problems, right?

Reply:
Actually fixing the tests for the additioal setpoins as in https://github.com/QCoDeS/Qcodes/pull/2259 makes it clear that they should be included to make the statement len(shapes) >= len(interdeps) true. So I have included them and added some explicit tests